tests/cunit/test_rearr.c

```diff
@@ -874,6 +874,98 @@ int test_box_rearrange_create_2(MPI_Comm test_comm, int my_rank)
     return 0;
 }
 
+/* Test for the box_rearrange_create() function.
+ * Test totalrecv > iodesc->llen case in compute_counts(). */
+int test_box_rearrange_create_3(MPI_Comm test_comm, int my_rank)
+{
+#define MAPLEN1 1
+    iosystem_desc_t *ios;
+    io_desc_t *iodesc;
+    io_region *ior1;
+    int maplen = MAPLEN1;
+    PIO_Offset compmap[MAPLEN1] = {0};
+    const int gdimlen[NDIM1] = {1};
```

Comment:
I believe this syntax does not accomplish what one might think. Only `{0}` will init the whole automatic array to 0. `{1}` will set the first entry only to 1 and leave the rest undefined.

Reply:
Thanks for pointing it out. This kind of initialization has more than 10 occurrences in test_rearr.c, which is based on the fact that NDIM1 is 1. 
```
/* For 1-D use. */
#define NDIM1 1
```